src/package-manager.coffee

```diff
@@ -409,6 +412,41 @@ class PackageManager
     message = "Failed to load the #{path.basename(packagePath)} package"
     atom.notifications.addError(message, {stack, detail, dismissable: true})
 
+  # TODO: remove these autocomplete-plus specific helpers after a few versions.
+  uninstallAutocompletePlus: ->
+    packageDir = null
+    devDir = path.join("dev", "packages")
+    for packageDirPath in @packageDirPaths
+      if not packageDirPath.endsWith(devDir)
+        packageDir = packageDirPath
+        break
+
+    if packageDir?
+      dirsToRemove = [
+        path.join(packageDir, 'autocomplete-plus')
+        path.join(packageDir, 'autocomplete-atom-api')
+        path.join(packageDir, 'autocomplete-css')
+        path.join(packageDir, 'autocomplete-html')
+        path.join(packageDir, 'autocomplete-snippets')
+      ]
+      for dirToRemove in dirsToRemove
+        @uninstallDirectory(dirToRemove)
+    return
+
+  uninstallDirectory: (directory) ->
+    symlinkPromise = new Promise (resolve) ->
+      fs.isSymbolicLink directory, (isSymLink) -> resolve(isSymLink)
+
+    dirPromise = new Promise (resolve) ->
+      fs.isDirectory directory, (isDir) -> resolve(isDir)
+
+    Promise.all([symlinkPromise, dirPromise]).then (values) ->
+      [isSymLink, isDir] = values
+      if isSymLink
+        fs.unlink(directory)
```

Comment:
> I tried several times manually, and there is a spec that hits this.

This is what I tried from the dev tools.

![screen shot 2015-05-12 at 9 47 27 am](https://cloud.githubusercontent.com/assets/671378/7593029/f0cfafb6-f88b-11e4-915e-05c4eac899a2.png)